[DOCS] Remove references to bootstrap.system_call_filter

grcevski opened the pull request:
This PR removes all references to the now removed option of `bootstrap.system_call_filter`.

Closes #84862

Commit:
Remove references to bootstrap.system_call_filter

elasticmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

elasticmachine commented:
Pinging @elastic/es-docs (Team:Docs)

rjernst reviewed (commented):
I don’t think we can remove this currently in master, since it would be a breaking change? That is, removing the actual setting. We could still make the docs change.

grcevski commented:
I think the setting was removed in 8.0 (https://github.com/elastic/elasticsearch/pull/72848) and its removal is [listed in the breaking changes](https://www.elastic.co/guide/en/elastic-stack/8.0/elasticsearch-breaking-changes.html). This just removes some outstanding docs + references to now removed setting. The actual setting static I removed was flagged as unused by IntelliJ, nothing depends on it.

rjernst reviewed (approved):
Ah, LGTM then!

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [8.0](https://github.com/elastic/elasticsearch/pull/85971)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/85971">](https://github.com/elastic/elasticsearch/pull/85971) |
| ✅ |  [8.2](https://github.com/elastic/elasticsearch/pull/85972)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/85972">](https://github.com/elastic/elasticsearch/pull/85972) |
| ✅ |  [8.1](https://github.com/elastic/elasticsearch/pull/85973)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/85973">](https://github.com/elastic/elasticsearch/pull/85973) |
